### PR TITLE
Extract if/elif checks from DQC 5

### PR DESCRIPTION
**Pull request by jasonleinbach-wf**
#### Changes:

- Extracted big check to function.
- Added tests.

Please review: @hefischer  @andrewperkins-wf

### CODE REVIEW

**Comment by brianmyers-wf** [comment]
+1

**Comment by andrewperkins-wf** [comment]
+1

File: dqc_us_rules/dqc_us_0005.py [hunk 1]
```diff
@@ -97,44 +97,64 @@ def validate_facts(val):
...
+                '{base_code}.17'.format(
+                    base_code=_CODE_NAME
+                ),
+                messages.get_message(_CODE_NAME, "17"),
+                modelObject=[fact] +
+                list(eop_results[lookup]),
+                ruleVersion=_RULE_VERSION
+            )
+    elif facts.axis_exists(
+            val, fact, 'SubsequentEventTypeAxis'):
```
Comment (chrislococo-wf) [inline dqc_us_rules/dqc_us_0005.py]: I would close this `):` on a newline to keep it cleaner

File: dqc_us_rules/dqc_us_0005.py [hunk 2]
```diff
@@ -97,44 +97,64 @@ def validate_facts(val):
...
+            '{base_code}.48'.format(base_code=_CODE_NAME),
+            messages.get_message(_CODE_NAME, "48"),
+            modelObject=[fact] + list(eop_results[lookup]),
+            ruleVersion=_RULE_VERSION
+        )
+    elif facts.axis_member_exists(
+            val,
+            fact,
+            'StatementScenarioAxis',
+            'ScenarioForecastMember'):
```
Comment (chrislococo-wf) [inline dqc_us_rules/dqc_us_0005.py]: Ditto on the closing `):` on a newline

File: dqc_us_rules/dqc_us_0005.py [hunk 3]
```diff
@@ -97,44 +97,64 @@ def validate_facts(val):
...
+        val.modelXbrl.error(
+            '{base_code}.49'.format(base_code=_CODE_NAME),
+            messages.get_message(_CODE_NAME, "49"),
+            modelObject=[fact] + list(eop_results[lookup]),
+            ruleVersion=_RULE_VERSION)
```
Comment (chrislococo-wf) [inline dqc_us_rules/dqc_us_0005.py]: SG: Please close context on new line.

File: tests/unit_tests/test_dqc_us_0005.py [hunk 4]
```diff
@@ -128,6 +128,68 @@ def test_axis_exists_multi_axis(self):
             facts.axis_exists(mock_val, mock_fact, 'not a used axis')
         )
 
+    def test_run_checks_EntityCommonStockSharesOutstanding(self):
+        msg = messages.get_message('DQC.US.0005', "17")
```
Comment (chrislococo-wf) [inline tests/unit_tests/test_dqc_us_0005.py]: SG: Single line testing docstring missing. ([REF](https://github.com/Workiva/Stylin/blob/master/STYLEGUIDE.rst#docstrings-and-unit-tests))

File: tests/unit_tests/test_dqc_us_0005.py [hunk 5]
```diff
@@ -128,6 +128,68 @@ def test_axis_exists_multi_axis(self):
...
+        mock_error.assert_called_with(
+            'DQC.US.0005.17',
+             msg,
+            modelObject=[fact] + list(eop_results[lookup]),
+            ruleVersion=dqc_us_0005._RULE_VERSION
+        )
+
+    @patch('dqc_us_rules.dqc_us_0005.facts.axis_exists')
+    def test_run_checks_axis_exists(self, axis_exists):
+        axis_exists.return_value = True
```
Comment (chrislococo-wf) [inline tests/unit_tests/test_dqc_us_0005.py]: SG: Single line testing docstring missing. ([REF](https://github.com/Workiva/Stylin/blob/master/STYLEGUIDE.rst#docstrings-and-unit-tests))

File: tests/unit_tests/test_dqc_us_0005.py [hunk 6]
```diff
@@ -128,6 +128,68 @@ def test_axis_exists_multi_axis(self):
...
+            'DQC.US.0005.48',
+            msg,
+            modelObject=[fact] + list(eop_results[lookup]),
+            ruleVersion=dqc_us_0005._RULE_VERSION
+        )
+
+    @patch('dqc_us_rules.dqc_us_0005.facts.axis_exists')
+    @patch('dqc_us_rules.dqc_us_0005.facts.axis_member_exists')
+    def test_run_checks_axis_member_exists(self, axis_member_exists, axis_exists):
+        axis_exists.return_value = False
```
Comment (chrislococo-wf) [inline tests/unit_tests/test_dqc_us_0005.py]: SG: Single line testing docstring missing. ([REF](https://github.com/Workiva/Stylin/blob/master/STYLEGUIDE.rst#docstrings-and-unit-tests))

**Comment by chrislococo-wf** [comment]
Followup for style comments made here: https://jira.atl.workiva.net/browse/XS-3018

**Comment by chrislococo-wf** [comment]
#### QA Steps: 

- Manual testing: Manually tested that 0005 still fires.
- Flake8 on dqc_us_rules: 
```
(dqc)  chrislococo [dqc] (XS-2905) $ flake8 dqc_us_rules
dqc_us_rules/dqc_us_0005.py:111:80: E501 line too long (84 > 79 characters)
(dqc)  chrislococo [dqc] (XS-2905) $
```
- Flake8 on tests: 
```
(dqc)  chrislococo [dqc] (XS-2905) $ flake8 tests/
tests/unit_tests/test_dqc_us_0005.py:134:80: E501 line too long (89 > 79 characters)
tests/unit_tests/test_dqc_us_0005.py:136:30: E231 missing whitespace after ':'
tests/unit_tests/test_dqc_us_0005.py:143:21: E225 missing whitespace around operator
tests/unit_tests/test_dqc_us_0005.py:144:80: E501 line too long (89 > 79 characters)
tests/unit_tests/test_dqc_us_0005.py:148:14: E131 continuation line unaligned for hanging indent
tests/unit_tests/test_dqc_us_0005.py:160:30: E231 missing whitespace after ':'
tests/unit_tests/test_dqc_us_0005.py:174:80: E501 line too long (82 > 79 characters)
tests/unit_tests/test_dqc_us_0005.py:181:30: E231 missing whitespace after ':'
(dqc)  chrislococo [dqc] (XS-2905) $
```
- Nosetest result: 
```
(dqc)  chrislococo [dqc] (XS-2905) $ nosetests tests/
.........................................................................
----------------------------------------------------------------------
Ran 73 tests in 0.039s

OK
(dqc)  chrislococo [dqc] (XS-2905) $
```

#### Result: QA +10, [follow up](https://jira.atl.workiva.net/browse/XS-3027) created for flake8 errors.

**Comment by andrewperkins-wf** [comment]
QA'ed and reviewed.  Followup tickets were made in our internal ticketing system to finish the styling of these changes.

**Pull request merged**